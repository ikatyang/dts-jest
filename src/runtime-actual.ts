import pretty_format = require('pretty-format');
import { runtime_indent_spaces, ActualResult } from './definitions';
import { indent } from './utils/indent';
import { safe_snapshot } from './utils/safe-snapshot';

export class ActualRuntime {
  public results: { [line: number]: ActualResult };

  constructor(results: { [line: number]: ActualResult }) {
    this.results = results;
  }

  public report(line: number, getter: () => any) {
    const result = this.results[line];
    const description =
      result.description === undefined ? '' : `\n${result.description}\n`;

    const expression = indent(result.expression, runtime_indent_spaces);
    const actual_value = indent(
      safe_snapshot(getter, pretty_format),
      runtime_indent_spaces,
    );

    return `${description}\nEvaluated\n\n${expression}\n\nto be\n\n${actual_value}\n`;
  }
}
