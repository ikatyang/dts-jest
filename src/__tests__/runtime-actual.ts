import { ActualRuntime } from '../runtime-actual';

const enum Line {
  Normal,
  Description,
}

let runtime: ActualRuntime;
beforeEach(() => {
  runtime = new ActualRuntime({
    [Line.Normal]: {
      line: Line.Normal,
      expression: 'expression',
      value: 'does not matter',
    },
    [Line.Description]: {
      line: Line.Description,
      description: 'description',
      expression: 'expression',
      value: 'does not matter',
    },
  });
});

test('#report() should return correctly', () => {
  expect(runtime.report(Line.Normal, () => 'actual-value')).toMatchSnapshot();
});

test('#report() should return correctly with description', () => {
  expect(
    runtime.report(Line.Description, () => 'actual-value'),
  ).toMatchSnapshot();
});
