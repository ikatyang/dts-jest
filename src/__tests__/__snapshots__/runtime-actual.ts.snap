// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#report() should return correctly 1`] = `
"
Evaluated

  expression

to be

  \\"actual-value\\"
"
`;

exports[`#report() should return correctly with description 1`] = `
"
description

Evaluated

  expression

to be

  \\"actual-value\\"
"
`;
